### PR TITLE
Re-enables automatic refresh

### LINKED ISSUE #53009: Fix apiRequest bug
## Description
A developer within the MHV Secure Messages migration team noted that `apiRequests` under certain conditions were failing unexpectedly as [described in this thread](https://dsva.slack.com/archives/CBU0KDSB1/p1674665827377169).  The issue is that `.json()` Response method is called inappropriately for various content-types that do not match.

## Tasks
- [ ] Fix the `.json()` failed calls from failing by having a backup
- [ ] Create unit tests for the `apiRequest` to test functionality

### PR DESCRIPTION
**Pull request by asg5704**
## Summary

- *(Summarize the changes that have been made to the platform)*
This was a bug in the `retryOn` method in the `apiRequest` utility function. In which attempting to parse the response using `.json()` resulted in an error for non-JSON responses (eg PDFs, 204 No Content).

- *(If bug, how to reproduce)*
https://dsva.slack.com/archives/C5AGLBNRK/p1675352034193369 (Attempt to download a letter)
https://dsva.slack.com/archives/CBU0KDSB1/p1674665827377169 (MHV Secure Messaging with 204 response code)

- *(What is the solution, why is this the solution)*
The solution is only to call the `.json()` Promise when a status code `403` is instead of on every response.
  - This fixes the `204 No Content` response as mentioned by the MHV Secure Messaging team
  - This fixes any non-JSON response (eg Letter Download from having a `.json()` arbitrarily called on it)

- *(Which team do you work for, does your team own the maintenance of this component?)*
Identity

- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*
There is not Flipper for this however, it will not be released to Production.

## Related issue(s)
Closes department-of-veterans-affairs/va.gov-team#53009
- *Link to ticket created in va.gov-team repo*
- *Link to previous change of the code/bug (if applicable)*
https://github.com/department-of-veterans-affairs/vets-website/pull/23011
- *Link to epic if not included in ticket*

## Testing done

- *Describe what the old behavior was prior to the change*
Background: The old behavior failed the PDF download when attempting to parse the response using `.json()` due to the error checking mechanism within `retryOn` method.  Another bug was found that stemmed from the `isomorphic-fetch` package that disallowed cloning of responses using the `.clone()` method.

- *Describe the steps required to verify your changes are working as expected*
1. Set up vets-api following these instructions: https://dsva.slack.com/archives/C5AGLBNRK/p1675374081359829?thread_ts=1675352034.193369&cid=C5AGLBNRK
2. Start the Frontend using `yarn watch`
3. Sign in `oauth=false` using ID.me (user+228)
4. After authenticating navigate to `http://localhost:3001/records/download-va-letters/letters/letter-list`
5. Chose any letter in the letter list and click Download
6. Confirm the PDF downloads to the users computer

- *Describe the tests completed and the results*
Creation of unit tests for the `apiRequest` functionality for 200, 204 (Alex Morgun requested), 400, and file download

## Screenshots
n/a

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
The entire site is impacted, please test accordingly

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Confirm the following
---

#### 🔴 VFS Teams
1. Set up vets-api up and running (based on your teams requirements)
2. Start vets-website by running `yarn watch`
3. Sign in with OAuth disabled (`oauth=false`) and test your respective application
  a. Primarily conduct manual testing
  b. Also includes unit testing & E2E Cypress testing
4. Sign in with OAuth enabled (this is enabled by default `oauth=true`)
  a. Primarily conduct manual testing
  b. Also conduct unit & E2E Cypress testing
  c. Wait 6 minutes for the Access Token to expire (any time after the `atExpires` time in Local Storage)
  d. After those 6 minutes continue manually testing
    1. Did you see a `403` response followed by a `/refresh` call?
    2. Was there anything unexpected?

--- 
#### ✅ Letters team (complete)
1. Set up vets-api following these instructions: https://dsva.slack.com/archives/C5AGLBNRK/p1675374081359829?thread_ts=1675352034.193369&cid=C5AGLBNRK
2. Start the Frontend using `yarn watch`
3. Sign in `oauth=false` using ID.me (user+228)
4. After authenticating navigate to `http://localhost:3001/records/download-va-letters/letters/letter-list`
5. Chose any letter in the letter list and click Download
6. Confirm the PDF downloads to the users computer

#### ✅ Unit tests + E2E tests
1. Unit tests work as expected
2. Cypress tests work as expected

### CODE REVIEW

File: src/platform/utilities/api/index.js [hunk 1]
```diff
@@ -23,10 +28,16 @@ export function fetchAndUpdateSessionExpiration(url, settings) {
     ...settings,
     async retryOn(attempt, error, response) {
       if (error) return false;
-      const atError = await response.clone().json();
+
+      const errorResponse = isJson
```
Comment (jerekshoe) [inline src/platform/utilities/api/index.js:32]: You might have caught this already, but shouldn't this be `isJson(response)`?

File: src/platform/utilities/api/index.js [hunk 2]
```diff
@@ -23,10 +28,16 @@ export function fetchAndUpdateSessionExpiration(url, settings) {
...
+        ? await response
+            ?.clone()
+            ?.json()
+            .catch(() => response.text())
+        : response.text();
```
Comment (jerekshoe) [inline src/platform/utilities/api/index.js:37]: You might need to clone this as well, I'm seeing an issue with the stream having already been read if not

**Review by jerekshoe — APPROVED** [review]
Works for the letters app

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Comment by jerekshoe** [comment]
Re-tested with the letters app and this works great!

**Review by acrollet — CHANGES_REQUESTED** [review]
I'm seeing an API request fail for us on the review environment for this PR that works correctly on staging. Digging into why, but leaving this here to prevent merge in the meantime. Thanks!

**Comment by asg5704** [comment]
> I'm seeing an API request fail for us on the review environment for this PR that works correctly on staging. Digging into why, but leaving this here to prevent merge in the meantime. Thanks!

Perfect thank you @acrollet, I wasn't going to roll this out until more VFS teams had time to test.

**Review by acrollet — APPROVED** [review]
Turns out the error was on my side earlier, the issue was with using Safari with review environments. Re-tested with Chrome and our check-in and pre-check-in workflows are working correctly.

**Pull request merged**